src/cli/java/org/commcare/util/screen/EntityScreen.java

```diff
@@ -131,7 +131,7 @@ public void init(SessionWrapper session) throws CommCareSessionException {
 
         evalContext.setQueryContext(newContext);
 
-        if (needsFullInit || references.size() == 1) {
+        if (needsFullInit || references.size() == 1 || shouldAutoSelect()) {
```

Comment:
Having the extra condition here doesn't makes any difference: `A or B or (C and B) = A or B`
(re-posting from https://github.com/dimagi/commcare-core/pull/1152#discussion_r1024935312)